### PR TITLE
update parser for cal-ciso to enable historical refetch

### PR DESCRIPTION
**Pull request by mathilde-daugy**
## Issue

The parser for US-CAL-CISO did not collect data for past dates because the link in `fetch_historical_production` was broken.
image

## Description

The parser has been updated to:

- parse data from past dates using the proxy and the same page as the real-time data.
- parse demand data

### Preview

demand 
image

production
image

### Double check

- [ ] I have tested my parser changes locally with `poetry run test_parser "zone_key"`
- [ ] I have run `pnpx prettier --write .` and `poetry run format` to format my changes.

### CODE REVIEW

File: parsers/US_CA.py [hunk 1]
```diff
@@ -26,159 +56,91 @@ def fetch_production(
...
-                    ca_gen_type
-                    + " production for US_CA was reported as less than 0 and was clamped"
-                )
-                production = 0.0
-
-            # if another mean of production created a value, sum them up
-            data["production"][mapped_gen_type] += production
+@refetch_frequency(timedelta(days=1))
+def fetch_consumption(
+    zone_key: str = "US-CA",
```
Comment (unitrium) [inline parsers/US_CA.py:107]:
```suggestion
    zone_key: str = "US-CAL-CISO",
```
I'm not even sure there should be a default key here.

File: parsers/US_CA.py [hunk 2]
```diff
@@ -256,10 +215,7 @@ def fetch_exchange(
     from pprint import pprint
 
     print("fetch_production() ->")
-    pprint(fetch_production())
```
Comment (unitrium) [inline parsers/US_CA.py:259]: I recommend leaving those as examples.

File: parsers/US_CA.py [hunk 3]
```diff
@@ -256,10 +215,7 @@ def fetch_exchange(
...
+    # pprint(fetch_production(target_datetime=datetime(2023,1,20)))s
+    pprint(fetch_consumption(target_datetime=datetime(2022,2,22)))
 
-    print('fetch_exchange("US-CA", "US") ->')
```
Comment (unitrium) [inline parsers/US_CA.py:261]: Same here, they might be useful for someone to try the parser out.

**Review by unitrium — APPROVED** [review]
A few minor things but I can already approve and this can be merged afterwards.

File: parsers/US_CA.py [hunk 4]
```diff
@@ -6,17 +6,47 @@
...
+    "Nuclear": "nuclear",
+    "Natural Gas": "gas",
+    "Large Hydro": "hydro",
+    "Other": "unknown",
+}
+STORAGE_MAPPING = {"Batteries": "battery"}
 
 MX_EXCHANGE_URL = "http://www.cenace.gob.mx/Paginas/Publicas/Info/DemandaRegional.aspx"
 
+def get_target_url(target_datetime:datetime, kind: str)->str:
```
Comment (VIKTORVAV99) [inline parsers/US_CA.py:43]:
```suggestion
def get_target_url(target_datetime: Optional[datetime], kind: str)->str:
```

File: parsers/US_CA.py [hunk 5]
```diff
@@ -26,159 +56,91 @@ def fetch_production(
...
+    zone_key: str = "US-CA",
+    session: Optional[Session] = None,
+    target_datetime: Optional[datetime] = None,
+    logger: Logger = getLogger(__name__),
+) -> list:
+    """Requests the last known production mix (in MW) of a given country."""
 
-        for ca_storage_type, mapped_storage_type in storage_map.items():
-            storage = -float(csv[ca_storage_type][i])
+    target_url = get_target_url(target_datetime, kind="consumption")
```
Comment (VIKTORVAV99) [inline parsers/US_CA.py:114]: Could we maybe rename these to better indicate they return CSV?

File: parsers/US_CA.py
Comment (VIKTORVAV99) [inline parsers/US_CA.py:1]: The defaultdict import is no longer needed.

File: parsers/US_CA.py (hunk 5 above)
Comment (mathilde-daugy) [reply parsers/US_CA.py:114]: it returns a link for the csv ie `'https://us-ca-proxy-jfnx5klx2a-uw.a.run.app/outlook/SP/History/20230412/netdemand.csv?host=https://www.caiso.com'`

**Pull request merged**